### ISSUE TITLE
How to to install in Raspberry Pi 2 or another plataform

### ISSUE THREAD

**Issue by adiogarda** [Brewskey/spark-server#115]
Hi,

I have a project in my company, and we need to use a local cloud for Particle Photons.
Because need all time the availability of exchange messagens between some Photons.

All how to in the internet for install in Raspberry Pi (2) not work for Spark Server.

I need know a simple but functionall how to for install in Raspberry Pi 2 or in any plataform (recommended) .
And I need know if need just the Spark Server installed, or Spark Protocol  (Particle).
Some erros I have encontered is about node deprecated functions. All not explain the exact version for this actual Spark Server.

Thanks 
Adio

**Comment by jlkalberer** [comment]
Hi,

We haven't done any testing with the Rasberry Pi. Those instructions were for the `Particle` fork of the library but I would assume that those instructions should still "work" to some extent.

You'll need to make sure you install the `particle-cli` instead of the older `spark-cli`.

Anyway, I just ordered a Rasberry Pi 3 so I'll be able to test this out.
----

The main instructions for setting up the server on Win/OSX/Linux works. 

Don't worry about the deprecation errors.  The server should still run fine

All you need to install is Spark Server and it will install the Spark Protocol.

**Comment by adiogarda** [comment]
Hi,

Thanks for your fast response and help.
Your project is great. And have a very important use in the ecosystem of Particle.
For use in some automation project, a high number of companies not agree the machines have connection of internet.
My case is this.
I need know more for finish install in Windows:
1 - Whats recommended version of Windows.... and x86 or x64 all softwares ?
2 - What's your version of Visual Studio ? and need this ?
3 - In my case now, the URSA not complete the compile ... I believe because the version of Visual Studio...

If you port your latest version with a easy and clean installation in Raspberry Pi 3 or UDOO NEO I contribute with your project. This two hardwares is cheap and very usefull in others tasks in automation.

Thanks,
Adio

**Comment by jlkalberer** [comment]
1. Any version of windows works
2. No need for visual studio - install windows build tools https://github.com/nodejs/node-gyp  I'll add this instruction to this repo

**Comment by jlkalberer** [comment]
@adiogarda - I've updated the Pi install instructions.  There were a few issues with installing node and particle-cli but everything else worked.

**Comment by adiogarda** [comment]
Hi John Kalberer,

Many thanks for all info.
As mentioned, your development and project is very important for the community of Particle.

My report status for your project:

Windows 2012 --> Worked fine all (spark-protocol, spark-server, particle-cli).

With (node version 6.9.5 - npm version 3.10.10)
Raspberry Pi B+ --> latest raspbian - worked fine
Udoo Neo --> latest udobuntu - worked fine

UBNT (Ubiquiti) Cloud Key (version 5.4.11) - Worked Fine, but need two things:
* Change the port 8080 for another (ex: 8090), because in the 8080 have the another web service.
* Install in SD CARD (/srv), because the space in the main storage is small.

For me and some, I install automation with local WiFi (not external access) and all of my time I use the UBNT products (Cloud key, switches, Access Points, etc...) all time I install the Cloud Key device (controller) and this device use Debian.

Thanks,
Adio - Spark

**Comment by jlkalberer** [comment]
Thanks. That's great to hear.

I also ran into some trouble with the port -- you should be able to pass `-p 8090` as a parameter to node to change the port.

**Comment by adiogarda** [comment]
Hi John,

I try to claim the Photon in my Local Cloud (server). But in the past this devices I used in Particle Cloud.
I have executed in Particle Cloud the unclaim..

But I have the error: "Unclaimed Device" in the log of server.
And In the moment I start the server (main.js) I have the "Access token: null"  .... Is a error ?
If I try to connect the Photon in Particle Cloud the access token is valid.. but not in my server...

What's wrong ?

Thanks
Adio

**Comment by jlkalberer** [comment]
You need to follow the main readme instructions. `particle keys doctor device_id` while in DFU mode.

I couldn't get it to work while plugging the photon into the Raspberry Pi.

**Comment by adiogarda** [comment]
Hi John,

I runned a connection test scenario using two photons. 
One is publishing 500 numbers (sequentially, from 1 to 500) at a given frequency. The frequency I inform via serial during startup. 
The other photon is subscribing to the first. 
In both I count the cloud connection losses. 
In the second I also count the delay between arrived messages and the numbers losts (probably due to longer cloud disconnection). 

As result: (a) using 300ms of interval, the second photon (subscription) reconnected to cloud 4 times; (b) at 500ms interval, the second photon reconnected to cloud 8 times and; (c) at 1000ms interval, the second photon reconnected to cloud 15 times. Based on these values, I imagined that the server is dropping (ending session for) the second (subscribed) photon at every half minute. 

Is that behaviour expected?
